cmd/server/search_handlers.go

```diff
@@ -83,163 +89,265 @@ func searchFEDACH(logger log.Logger, searcher *searcher) http.HandlerFunc {
 			moovhttp.Problem(w, errNoSearchParams)
 		}
 
-		// Search by Name Only
 		if req.nameOnly() {
 			if logger != nil {
 				logger.Log("searchFEDACH", fmt.Sprintf("searching FED ACH Dictionary by name only %s", req.Name))
 			}
-			req.searchNameOnly(logger, searcher)(w, r)
+			req.searchNameOnly(logger, searcher, ACH)(w, r)
 			return
 		} else if req.routingNumberOnly() {
 			if logger != nil {
 				logger.Log("searchFEDACH", fmt.Sprintf("searching FED ACH Dictionary by routing number only %s", req.RoutingNumber))
 			}
-			req.searchRoutingNumberOnly(logger, searcher)(w, r)
+			req.searchRoutingNumberOnly(logger, searcher, ACH)(w, r)
 			return
 		} else if req.stateOnly() {
 			if logger != nil {
 				logger.Log("searchFEDACH", fmt.Sprintf("searching FED ACH Dictionary by state only %s", req.State))
 			}
-			req.searchStateOnly(logger, searcher)(w, r)
+			req.searchStateOnly(logger, searcher, ACH)(w, r)
 			return
 		} else if req.cityOnly() {
 			if logger != nil {
 				logger.Log("searchFEDACH", fmt.Sprintf("searching FED ACH Dictionary by city only %s", req.City))
 			}
-			req.searchCityOnly(logger, searcher)(w, r)
+			req.searchCityOnly(logger, searcher, ACH)(w, r)
 			return
 		} else if req.postalCodeOnly() {
 			if logger != nil {
 				logger.Log("searchFEDACH", fmt.Sprintf("searching FED ACH Dictionary by postal code only %s", req.PostalCode))
 			}
-			req.searchPostalCodeOnly(logger, searcher)(w, r)
+			req.searchPostalCodeOnly(logger, searcher, ACH)(w, r)
 			return
 		} else {
 			if logger != nil {
 				logger.Log("searchFEDACH", fmt.Sprintf("searching FED ACH Dictionary by parameters %v", req.RoutingNumber))
 			}
-			req.searchACH(logger, searcher)(w, r)
+			req.search(logger, searcher, ACH)(w, r)
 			return
 		}
-
 	}
 }
 
-// searchNameOnly searches FEDACH by name only
-func (req fedachSearchRequest) searchNameOnly(logger log.Logger, searcher *searcher) http.HandlerFunc {
+// searchNameOnly searches FEDACH / FEDWIRE by name only
+func (req fedSearchRequest) searchNameOnly(logger log.Logger, searcher *searcher, searchType string) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		if logger != nil {
 			logger.Log("searchFEDACH", fmt.Sprintf("search by name %s", req.Name))
 		}
 
-		achP, err := searcher.ACHFindNameOnly(req.Name)
-		if err != nil {
-			moovhttp.Problem(w, err)
-		}
-
-		w.Header().Set("Content-Type", "application/json; charset=utf-8")
-		w.WriteHeader(http.StatusOK)
-		if err := json.NewEncoder(w).Encode(&searchResponse{ACHParticipants: achP}); err != nil {
-			moovhttp.Problem(w, err)
-			return
+		switch searchType {
+		case ACH:
+			achP, err := searcher.ACHFindNameOnly(req.Name)
+			if err != nil {
+				moovhttp.Problem(w, err)
+			}
+			w.Header().Set("Content-Type", "application/json; charset=utf-8")
+			w.WriteHeader(http.StatusOK)
+			if err := json.NewEncoder(w).Encode(&searchResponse{ACHParticipants: achP}); err != nil {
+				moovhttp.Problem(w, err)
+				return
+			}
```

Comment:
We could have all of these switch statements return/set a searchResponse instead of duplicating the HTTP response code.

Reply:
yeah that's fine